File: 프로그래머스/unrated/181881. 조건에 맞게 수열 변환하기 2/조건에 맞게 수열 변환하기 2.js
function solution(arr) {
    let x = 0;
    let temp = [];
    
    while (true) {
        temp = [...arr];
        
        arr.forEach((el, i) => {
            if (el % 2 === 0 && el >= 50) {
                arr[i] /= 2;
            } else if (el % 2 === 1 && el < 50) {
                arr[i] *= 2;
                arr[i] += 1;
            }
        });
        
        if (JSON.stringify(temp) === JSON.stringify(arr)) {
            break;
        }
        
        x++;
    }
    
    return x;
}

File: 프로그래머스/unrated/181881. 조건에 맞게 수열 변환하기 2/조건에 맞게 수열 변환하기 2.java
import java.util.Arrays;

class Solution {
    public int solution(int[] arr) {
        int[] transform = Arrays.copyOfRange(arr, 0, arr.length);
        int count = 0;
        
        while (true) {
            for (int i = 0; i < arr.length; i++) {
                int num = arr[i];
                boolean isEven = num % 2 == 0;
                
                if (num >= 50 && isEven) {
                    num /= 2;
                } else if (num < 50 && !isEven) {
                    num *= 2;
                    num++;
                }
                arr[i] = num;
            }
            
            boolean stop = true;
            for (int i = 0; i < arr.length; i++) {
                if (transform[i] != arr[i]) {
                    stop = false;
                    break;
                }
            }
            
            if (stop) {
                return count;
            }
            
            transform = Arrays.copyOfRange(arr, 0, arr.length);
            
            count++;
        }
    }
}